include/EASTL/sort.h

```diff
@@ -205,77 +205,144 @@ namespace eastl
 
 
 
-	/// insertion_sort
-	///
-	/// Since insertion_sort requires that the data be addressed with a BidirectionalIterator and 
-	/// not the more flexible RandomAccessIterator, we implement the sort by doing a for loop within
-	/// a for loop. If we were to specialize this for a RandomAccessIterator, we could replace the
-	/// inner for loop with a call to upper_bound, which would be faster.
-	///
-	template <typename BidirectionalIterator, typename StrictWeakOrdering>
-	void insertion_sort(BidirectionalIterator first, BidirectionalIterator last, StrictWeakOrdering compare)
+	/* Insertion Sort
+	 *
+	 * An in-place and stable sorting algorithm that is fast for small arrays,
+	 * making it useful in algorithms like Quick Sort, Merge Sort, most-significant-
+	 * digit Radix Sorts, and Quick Select, which divide their inputs into smaller (semi)sorted segments.
+	 *
+	 * An improvement for larger inputs would be to use upper_bound to find the insertion position,
+	 * followed by memmove, but it will still be slow for large inputs, and not worth it for small ones.
+	 *
+	 * If base[-1] is a sentinel the short-circuit can be removed, but this is a general purpose version.
+	 */
+	
+	//compare functor version
+	template<class Iter, class Compare>//Bidirectional Iterator, though something like a list shouldn't use this
+	void insertion_sort(Iter const base, Iter const past, Compare cmp)
```

Comment:
The template parameter names and function parameters are named fairly consistently across the code base.  Please avoid renaming them unless it can't be avoided.